mergify_engine/web/api/eventlogs.py

```diff
@@ -0,0 +1,77 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2022 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import dataclasses
+import typing
+
+import daiquiri
+import fastapi
+import pydantic
+
+from mergify_engine import context
+from mergify_engine import eventlogs
+from mergify_engine import github_types
+from mergify_engine.web import api
+from mergify_engine.web.api import security
+
+
+LOG = daiquiri.getLogger(__name__)
+
+
+router = fastapi.APIRouter(
+    tags=["eventlogs"],
+    dependencies=[
+        fastapi.Depends(security.require_authentication),
+    ],
+)
+
+
+Event = typing.Annotated[
+    typing.Union[eventlogs.EventLabel, eventlogs.EventRefresh],
+    pydantic.Field(discriminator="event"),
+]
+
+
+@pydantic.dataclasses.dataclass
+class EventLogsResponse:
+    events: typing.List[Event] = dataclasses.field(
+        default_factory=list,
+        metadata={
+            "description": "The list of events of a pull request",
+        },
+    )
+
+
+@router.get(
+    "/repos/{owner}/{repository}/pulls/{pull}/events",  # noqa: FS003
+    summary="Get the event logs of a pull request",
+    description="Get the event logs of the requested pull request",
+    response_model=EventLogsResponse,
+    dependencies=[fastapi.Depends(security.check_subscription_feature_eventlogs)],
+    responses={
+        **api.default_responses,  # type: ignore
+    },
+)
+async def get_pull_request_eventlogs(
+    repository_ctxt: context.Repository = fastapi.Depends(  # noqa: B008
+        security.get_repository_context
+    ),
+    pull: github_types.GitHubPullRequestNumber = fastapi.Path(  # noqa: B008
+        ..., description="Pull request number"
+    ),
+) -> EventLogsResponse:
+    return EventLogsResponse(
+        events=[e async for e in eventlogs.get(repository_ctxt, pull)]
+    )
```

Comment:
What if we want both? 🤔

Reply:
I don't think we want an "advanced API".
Getting the N last events for a PR is enough, and if any filtering is needed, it can be done client-side (or server-side). The payload is not going to be gigantic.

Until we build a hell of a dashboard that supports advanced querying and display, we want the API to be mostly pass-through.